device_calendar/android/src/main/kotlin/com/builttoroam/devicecalendar/DeviceCalendarPlugin.kt

```diff
@@ -96,6 +104,22 @@ class DeviceCalendarPlugin() : MethodCallHandler {
                 event.start = eventStart!!
                 event.end = eventEnd!!
 
+                if(call.hasArgument(RECURRENCE_RULE_ARGUMENT) && call.argument<Map<String, Any>>(RECURRENCE_RULE_ARGUMENT) != null) {
+                    val recurrenceRuleArgs = call.argument<Map<String, Any>>(RECURRENCE_RULE_ARGUMENT)!!
+                    val recurrenceFrequencyIndex = recurrenceRuleArgs[RECURRENCE_FREQUENCY_ARGUMENT] as Int
+                    val recurrenceRule = RecurrenceRule(RecurrenceFrequency.values()[recurrenceFrequencyIndex])
+                    if(recurrenceRuleArgs.containsKey(TOTAL_OCCURRENCES_ARGUMENT)) {
+                        recurrenceRule.totalOccurrences = recurrenceRuleArgs[TOTAL_OCCURRENCES_ARGUMENT] as Int
+                    }
+                    if(recurrenceRuleArgs.containsKey(INTERVAL_ARGUMENT)) {
+                        recurrenceRule.interval = recurrenceRuleArgs[INTERVAL_ARGUMENT] as Int
+                    }
+                    if (recurrenceRuleArgs.containsKey(END_DATE_ARGUMENT)) {
+                        recurrenceRule.endDate = recurrenceRuleArgs[END_DATE_ARGUMENT] as Long
+                    }
+                    event.recurrenceRule = recurrenceRule
```

Comment:
Formatting: add spacing after curly braces? Not sure what's the styling guide for kotlin or generally what and if we agreed on any code styling

Reply:
we haven't but will do so